File: pueblo-golf-tournament-api/Controllers/V1/ManageController.cs
using Microsoft.AspNetCore.Mvc;
using pueblo_golf_tournament_api.Dto.Incoming;
using pueblo_golf_tournament_api.Dto.Outgoing;
using pueblo_golf_tournament_api.Modules.Manage;

namespace pueblo_golf_tournament_api.Controllers.V1
{
    [ApiController]
    [ApiVersion("1.0")]
    [Route("api/v{version:apiVersion}/manage")]
    public class ManageController : ControllerBase
    {

        private readonly IManageModule _manageModule;

        public ManageController(IManageModule manageModule)
        {
            _manageModule = manageModule;
        }

        [HttpPost("update/person")]
        public async Task<ActionResult<UpdatePersonResponseDto>> UpdatePerson(UpdatePersonRequestDto dto)
        {
            var updated = await _manageModule.UpdatePerson(dto.Person);

            var response = new UpdatePersonResponseDto
            {
                Data = updated,
                Message = updated != null ? "Person is successfully updated." : "Failed to update info"
            };

            return response != null ? Ok(response) : BadRequest(response);
        }
        [HttpPost("update/player")]
        public async Task<ActionResult<UpdatePlayerResponseDto>> UpdatePlayer(UpdatePlayerRequestDto dto)
        {
            var updated = await _manageModule.UpdatePlayer(dto.Player);

            var response = new UpdatePlayerResponseDto
            {
                Data = updated,
                Message = updated != null ? "Player is successfully updated." : "Failed to update info. Please try again."
            };

            return response != null ? Ok(response) : BadRequest(response);
        }
        [HttpPost("update/team")]
        public async Task<ActionResult<UpdateTeamResponseDto>> UpdateTeam(UpdateTeamRequestDto dto)
        {
            var updated = await _manageModule.UpdateTeam(dto.Team);

            var response = new UpdateTeamResponseDto
            {
                Data = updated,
                Message = updated != null ? "Team is successfully updated." : "Failed to update info. Please try again."
            };

            return response != null ? Ok(response) : BadRequest(response);
        }
        [HttpPost("update/tournament")]
        public async Task<ActionResult<UpdateTournamentResponseDto>> UpdateTournament(UpdateTournamentRequestDto dto)
        {
            var updated = await _manageModule.UpdateTournament(dto.Tournament);

            var response = new UpdateTournamentResponseDto
            {
                Data = updated,
                Message = updated != null ? "Tournament is successfully updated." : "Failed to update info. Please try again."
            };

            return response != null ? Ok(response) : BadRequest(response);
        }
        [HttpPost("update/tee-time-schedule")]
        public async Task<ActionResult<UpdateTeeTimeScheduleResponseDto>> UpdateTeeTimeSchedule(UpdateTeeTimeScheduleRequestDto dto)
        {
            var updated = await _manageModule.UpdateTeeTimeSchedule(dto.TeeTimeSchedule);

            if(updated != null){
                Console.WriteLine(updated.Id);
                Console.WriteLine(updated.TournamentId);
                Console.WriteLine(updated.DateTimeSlot.ToString());
                Console.WriteLine(updated.IsSelected);
                Console.WriteLine(updated.IsEnabled);
                Console.WriteLine(updated.IsFull);
            }

            var response = new UpdateTeeTimeScheduleResponseDto
            {
                Data = updated,
                Message = updated != null ? "Tee Time Schedule is successfully updated." : "Failed to update info. Please try again."
            };

            return response != null ? Ok(response) : BadRequest(response);
        }
        [HttpPost("confirm/payment")]
        public async Task<ActionResult<ConfirmPaymentResponseDto>> ConfirmPayment(ConfirmPaymentRequestDto dto)
        {
            var updated = await _manageModule.ConfirmPayment(dto);

            return updated != null ? Ok(updated) : BadRequest(updated);
        }
    }
}

File: pueblo_golf_tournament_mobile/lib/api/manage/manage-controller.dart
import 'dart:convert';

import 'package:get/get.dart';
import 'package:get/get_core/src/get_main.dart';
import 'package:pueblo_golf_tournament_mobile/api/manage/imanage-controller.dart';
import 'package:pueblo_golf_tournament_mobile/dto/request/confirm-payment-request-dto.dart';
import 'package:pueblo_golf_tournament_mobile/dto/request/update-person-request-dto.dart';
import 'package:pueblo_golf_tournament_mobile/dto/request/update-player-request-dto.dart';
import 'package:pueblo_golf_tournament_mobile/dto/request/update-team-request-dto.dart';
import 'package:pueblo_golf_tournament_mobile/dto/request/update-tee-time-schedule-request-dto.dart';
import 'package:pueblo_golf_tournament_mobile/dto/request/update-tournament-request-dto.dart';
import 'package:pueblo_golf_tournament_mobile/dto/response/confirm-payment-response-dto.dart';
import 'package:pueblo_golf_tournament_mobile/dto/response/update-person-response-dto.dart';
import 'package:pueblo_golf_tournament_mobile/dto/response/update-player-response-dto.dart';
import 'package:pueblo_golf_tournament_mobile/dto/response/update-team-response-dto.dart';
import 'package:pueblo_golf_tournament_mobile/dto/response/update-tee-time-schedule-response-dto.dart';
import 'package:pueblo_golf_tournament_mobile/dto/response/update-tournament-response-dto.dart';

import '../../utilities/http-controller.dart';

class ManageController extends IManageController {
  var controller = "/api/v1/manage";
  var httpController = Get.find<HttpController>();
  @override
  Future<UpdatePersonResponseDto> updatePerson(
      UpdatePersonRequestDto payload) async {
    var endpoint = "$controller/update/person";
    var response =
        await httpController.post(endpoint, jsonEncode(payload.toJson()));

    switch (response.statusCode) {
      case 200:
        return UpdatePersonResponseDto.fromJson(jsonDecode(response.body));
      case 400:
        return UpdatePersonResponseDto.fromJson(jsonDecode(response.body));
      default:
        return UpdatePersonResponseDto(
            message: response.reasonPhrase!, data: null);
    }
  }

  @override
  Future<UpdatePlayerResponseDto> updatePlayer(
      UpdatePlayerRequestDto payload) async {
    var endpoint = "$controller/update/player";
    var response =
        await httpController.post(endpoint, jsonEncode(payload.toJson()));

    switch (response.statusCode) {
      case 200:
        return UpdatePlayerResponseDto.fromJson(jsonDecode(response.body));
      case 400:
        return UpdatePlayerResponseDto.fromJson(jsonDecode(response.body));
      default:
        return UpdatePlayerResponseDto(
            message: response.reasonPhrase!, data: null);
    }
  }

  @override
  Future<UpdateTeamResponseDto> updateTeam(UpdateTeamRequestDto payload) async {
    var endpoint = "$controller/update/team";
    var response =
        await httpController.post(endpoint, jsonEncode(payload.toJson()));

    switch (response.statusCode) {
      case 200:
        return UpdateTeamResponseDto.fromJson(jsonDecode(response.body));
      case 400:
        return UpdateTeamResponseDto.fromJson(jsonDecode(response.body));
      default:
        return UpdateTeamResponseDto(
            message: response.reasonPhrase!, data: null);
    }
  }

  @override
  Future<UpdateTournamentResponseDto> updateTournament(
      UpdateTournamentRequestDto payload) async {
    var endpoint = "$controller/update/tournament";
    var response =
        await httpController.post(endpoint, jsonEncode(payload.toJson()));

    switch (response.statusCode) {
      case 200:
        return UpdateTournamentResponseDto.fromJson(jsonDecode(response.body));
      case 400:
        return UpdateTournamentResponseDto.fromJson(jsonDecode(response.body));
      default:
        return UpdateTournamentResponseDto(
            message: response.reasonPhrase!, data: null);
    }
  }

  @override
  Future<ConfirmPaymentResponseDto> confirmPayment(
      ConfirmPaymentRequestDto payload) async {
    var endpoint = "$controller/confirm/payment";
    var response =
        await httpController.post(endpoint, jsonEncode(payload.toJson()));

    switch (response.statusCode) {
      case 200:
        return ConfirmPaymentResponseDto.fromJson(jsonDecode(response.body));
      case 400:
        return ConfirmPaymentResponseDto.fromJson(jsonDecode(response.body));
      default:
        return ConfirmPaymentResponseDto(
            message: response.reasonPhrase!, payment: null, registration: null);
    }
  }

  @override
  Future<UpdateTeeTimeScheduleResponseDto> updateTeeTimeSchedule(
      UpdateTeeTimeScheduleRequestDto payload) async {
    var endpoint = "$controller/update/tee-time-schedule";
    var response =
        await httpController.post(endpoint, jsonEncode(payload.toJson()));

    switch (response.statusCode) {
      case 200:
        return UpdateTeeTimeScheduleResponseDto.fromJson(
            jsonDecode(response.body));
      case 400:
        return UpdateTeeTimeScheduleResponseDto.fromJson(
            jsonDecode(response.body));
      default:
        return UpdateTeeTimeScheduleResponseDto(
          message: response.reasonPhrase!,
          data: null,
        );
    }
  }
}
